bcdaworker/queueing/manager/que.go

```diff
@@ -107,6 +110,24 @@ func (q *queue) processJob(job *que.Job) error {
 		return nil
 	}
 
+	// start a goroutine that will periodically check the status of the parent job
+	go func() {
+		for {
+			select {
+			case <-time.After(15 * time.Second):
+				parentCancelled, _ := q.isParentJobCancelled(jobArgs.ID)
+				// parent job is cancelled, cancel this queuejob
+				if parentCancelled {
+					cancel()
+					return
```

Comment:
More code here in the future to stop the job?

Reply:
so cancelling here should mean it gets picked up by `worker.go#writeBBDataToFile` where it will stop and exit out of the loop, and return an error indicating that the parent job was cancelled. I'll add in some more comments to make those steps clearer!